iceoryx_binding_c/test/moduletests/test_trigger_state.cpp

```diff
@@ -0,0 +1,211 @@
+using namespace iox::popo;
+using namespace iox::capro;
+using namespace iox::mepoo;
+using namespace iox::cxx;
+using namespace iox::posix;
+
+
+extern "C" {
+#include "iceoryx_binding_c/subscriber.h"
+#include "iceoryx_binding_c/trigger_state.h"
+}
+
+#include "test.hpp"
+
+using namespace ::testing;
+
+class iox_trigger_state_test : public Test
+{
+  public:
+    const iox::capro::ServiceDescription TEST_SERVICE_DESCRIPTION{"a", "b", "c"};
+
+    void SetUp()
+    {
+        m_lastTriggerCallbackArgument = nullptr;
+        m_mempoolconf.addMemPool({CHUNK_SIZE, NUM_CHUNKS_IN_POOL});
+        m_memoryManager.configureMemoryManager(m_mempoolconf, &m_memoryAllocator, &m_memoryAllocator);
+        m_subscriber.m_portData = &m_portPtr;
+    }
+
+    void TearDown()
+    {
+    }
+
+    void Subscribe(SubscriberPortData* ptr)
+    {
+        uint64_t queueCapacity = MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY;
+        iox_sub_subscribe(m_subscriberHandle, queueCapacity);
+
+        SubscriberPortSingleProducer(ptr).tryGetCaProMessage();
+        iox::capro::CaproMessage caproMessage(iox::capro::CaproMessageType::ACK, TEST_SERVICE_DESCRIPTION);
+        SubscriberPortSingleProducer(ptr).dispatchCaProMessageAndGetPossibleResponse(caproMessage);
+    }
+
+    static void triggerCallback(UserTrigger* arg)
+    {
+        m_lastTriggerCallbackArgument = arg;
+    }
+
+
+    static UserTrigger* m_lastTriggerCallbackArgument;
+    ConditionVariableData m_condVar;
+    WaitSetMock m_waitSet{&m_condVar};
+    UserTrigger m_userTrigger;
+
+    static constexpr uint32_t NUM_CHUNKS_IN_POOL = MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY + 2;
+    static constexpr uint32_t CHUNK_SIZE = 128;
+    static constexpr size_t MEMORY_SIZE = 1024 * 1024 * 100;
+    uint8_t m_memory[MEMORY_SIZE];
+    Allocator m_memoryAllocator{m_memory, MEMORY_SIZE};
+    MePooConfig m_mempoolconf;
+    MemoryManager m_memoryManager;
+    iox::popo::SubscriberPortData m_portPtr{
+        TEST_SERVICE_DESCRIPTION, "myApp", iox::cxx::VariantQueueTypes::SoFi_SingleProducerSingleConsumer};
+    ChunkQueuePusher<SubscriberPortData::ChunkQueueData_t> m_chunkPusher{&m_portPtr.m_chunkReceiverData};
+    cpp2c_Subscriber m_subscriber;
+    iox_sub_t m_subscriberHandle = &m_subscriber;
+};
+
+UserTrigger* iox_trigger_state_test::m_lastTriggerCallbackArgument = nullptr;
+
+TEST_F(iox_trigger_state_test, triggerStateHasCorrectId)
+{
+    m_userTrigger.attachToWaitset(m_waitSet, 123, nullptr);
+    m_userTrigger.trigger();
+
+    auto triggerStateVector = m_waitSet.wait();
+    ASSERT_THAT(triggerStateVector.size(), Eq(1));
+
+    EXPECT_EQ(iox_trigger_state_get_trigger_id(&triggerStateVector[0]), 123);
+}
+
+TEST_F(iox_trigger_state_test, triggerOriginIsUserTriggerPointerWhenItsOriginatingFromThem)
+{
+    m_userTrigger.attachToWaitset(m_waitSet, 123, nullptr);
+    m_userTrigger.trigger();
+
+    auto triggerStateVector = m_waitSet.wait();
+    ASSERT_THAT(triggerStateVector.size(), Eq(1));
```

Comment:
You already checked this. So somehow a bit redundant

Reply:
You are right!